api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimService.java

```diff
@@ -106,19 +108,25 @@ private Claim buildClaim(Claimant claimant, EligibilityResponse eligibilityRespo
                 .build();
     }
 
-    private ClaimResult createResult(Claim claim, EligibilityResponse eligibilityResponse) {
-        Optional<VoucherEntitlement> entitlement = eligibilityResponse.getEligibilityStatus() == EligibilityStatus.ELIGIBLE
-                ? Optional.of(getEntitlement(claim, eligibilityResponse))
-                : Optional.empty();
+    private ClaimResult createResult(Claim claim) {
```

Comment:
I think you should combine the two `createResult` methods now and let them take in a `Optional(voucherEntitlement)`. If that doesn't quite work, then I'd inline this one as I don't think we want/need to have two overloaded methods here.

Reply:
I disagree, an optional parameter usually indicates that you have two different methods.